### PR TITLE
Feature/course

### PR DESCRIPTION
**Pull request by yagovaluchedevs**
## Description

Implementando o componente referente aos cursos que teremos no projeto.

## Dev Notes

- If any developer notes are required, list them here. If not, remove the 'Notes' section.

### Steps

1. List all steps in a numbered list

## Screenshots / Videos

Captura de Tela 2022-02-26 às 17 40 03

## Developer Checks

- [ ] PR title & commits adhere to [Conventional Commit Specification](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] PR is targeting correct branch, is up-to-date, & no merge conflicts
- [ ] Tested on browser device
- [ ] Tested on Mobile device

### CODE REVIEW

File: src/components/Courses/index.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import { Typography } from '..';
+import * as S from './styles';
+
+type PropsCourse = {
+  course: 'frontEnd' | 'backEnd' | 'design';
+};
+
+const COURSE_DATA = {
```
Comment (ramonxm) [inline src/components/Courses/index.tsx:9]: Acho válido mover isso pra uma pasta `mocks` ou `constants`

**Comment by jonataspinto** [comment]
<details>
    <summary>Lighthouse result for https://www.estartandodevs.com.br/</summary>
    

  | Metric | Score |  |
  | --- | --- | --- |
  | Performance | 0.5 | ✅ |
| First Contentful Paint | 3.8 s |  |
| Time to Interactive | 6.6 s |  |
| Speed Index | 5.1 s |  |
| Total Blocking Time | 800 ms |  |
| Largest Contentful Paint | 4.2 s |  |
| Cumulative Layout Shift | 0 |  |

    
</details>
<details>
    <summary>Lighthouse result for http://estartandodevs-site.s3-website-sa-east-1.amazonaws.com/</summary>
    

  | Metric | Score |  |
  | --- | --- | --- |
  | Performance | 0.96 | ✅ |
| First Contentful Paint | 1.5 s |  |
| Time to Interactive | 5.0 s |  |
| Speed Index | 2.2 s |  |
| Total Blocking Time | 150 ms |  |
| Largest Contentful Paint | 1.5 s |  |
| Cumulative Layout Shift | 0.001 |  |

    
</details>

**Review by jonataspinto — COMMENTED** [review]
| Device | URL |
|--|--|
| desktop | http://estartandodevs-site.s3-website-sa-east-1.amazonaws.com/ |

     
| Device | URL |
|--|--|
| mobile | http://estartandodevs-site.s3-website-sa-east-1.amazonaws.com/ |

Not what you expected? Are your scores flaky? [Run Lighthouse on Foo](https://www.foo.software/lighthouse)
If scores continue to be inconsistent consider [running all audits on Foo](https://github.com/foo-software/lighthouse-check-action#usage-foos-automated-lighthouse-check-api)

File: src/components/Courses/styles.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+const CourseContainer = styled.section`
+  width: 100%;
+  max-width: 1000px;
+  padding: 0 5px 0 60px;
+  margin-top: 100px;
+`;
+
+const TitleContainer = styled.div`
```
Comment (ramonxm) [inline src/components/Courses/styles.ts:15]: Poderia passar a prop no styled, styled.div<Prop> assim n precisa escrever duas vezes o type da prop

File: src/components/Courses/styles.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  > img {
+    cursor: pointer;
+  }
+`;
+
+const ListModulesContainer = styled.div`
+  margin: 10px 0 15px 0;
+
+  > p {
+    color: white;
```
Comment (ramonxm) [inline src/components/Courses/styles.ts:62]: Tem no tema essa aí

File: src/components/Courses/styles.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  max-width: ${({ width }: StyleProps) => width};
+  height: 23px;
+  position: relative;
+
+  > p {
+    position: absolute;
+    top: -13px;
+    left: 10px;
+    @media (max-width: 352px) {
+      font-size: 17px;
```
Comment (ramonxm) [inline src/components/Courses/styles.ts:26]: Vamos trabalhar com `rem` para responsividade dos textos

File: src/pages/index.tsx [hunk 5]
```diff
@@ -1,9 +1,12 @@
 import React from 'react';
-import { Layout, Footer, Header } from '../components';
+import { Layout, Footer, Header, Courses } from '../components';
 
 const Home = () => (
   <Layout>
     <Header />
+    <Courses course="frontEnd" />
```
Comment (ramonxm) [inline src/pages/index.tsx:7]: Poderia ser um array também com os nomes dos cursos, apesar de ser pequeno por enquanto acho que seria válido, mas é mais opinião mesmo

File: src/styles/global.tsx [hunk 6]
```diff
@@ -17,11 +17,6 @@ export const GlobalStyles = createGlobalStyle`
     scroll-behavior: smooth;
   }
 
-@media only screen and (max-width: 600px) {
```
Comment (ramonxm) [inline src/styles/global.tsx:20]: Qual a ideia? Isso foi colocado para controlar o tamanho dos textos no mobile

**Review by ramonxm — CHANGES_REQUESTED** [review]

File: src/styles/global.tsx (hunk 6 above)
Comment (ramonxm) [reply src/styles/global.tsx:20]: O rem é relativo ao tamanho do font-size, então eu defino de forma global o tamanho desse font-size, nesse caso faz sentido usar px e não rem. Pq na hora que eu utilizar o rem em um componente ele vai ser relativo ao tamanho de pixels que eu especifiquei

File: src/pages/index.tsx (hunk 5 above)
Comment (yagovaluchedevs) [reply src/pages/index.tsx:7]: Acho que seria alto esforço para pouco impacto.  Dessa forma escreveríamos mais código e não vejo necessidade de fato

File: src/components/Courses/styles.ts (hunk 3 above)
Comment (yagovaluchedevs) [reply src/components/Courses/styles.ts:62]: boa !!

File: src/components/Courses/styles.ts (hunk 4 above)
Comment (yagovaluchedevs) [reply src/components/Courses/styles.ts:26]: hummm, não sabia mano.. vou alterar aqui

File: src/components/Courses/index.tsx (hunk 1 above)
Comment (yagovaluchedevs) [reply src/components/Courses/index.tsx:9]: boa!!! faz total sentido

**Review by ramonxm — APPROVED** [review]

**Comment by jonataspinto** [comment]
<details>
    <summary>Lighthouse result for https://www.estartandodevs.com.br/</summary>
    

  | Metric | Score |  |
  | --- | --- | --- |
  | Performance | 0.55 | ✅ |
| First Contentful Paint | 3.4 s |  |
| Time to Interactive | 6.3 s |  |
| Speed Index | 4.1 s |  |
| Total Blocking Time | 500 ms |  |
| Largest Contentful Paint | 6.1 s |  |
| Cumulative Layout Shift | 0 |  |

    
</details>
<details>
    <summary>Lighthouse result for http://estartandodevs-site.s3-website-sa-east-1.amazonaws.com/</summary>
    

  | Metric | Score |  |
  | --- | --- | --- |
  | Performance | 0.96 | ✅ |
| First Contentful Paint | 1.6 s |  |
| Time to Interactive | 4.9 s |  |
| Speed Index | 2.5 s |  |
| Total Blocking Time | 100 ms |  |
| Largest Contentful Paint | 1.6 s |  |
| Cumulative Layout Shift | 0.044 |  |

    
</details>

**Review by jonataspinto — COMMENTED** [review]
| Device | URL |
|--|--|
| desktop | http://estartandodevs-site.s3-website-sa-east-1.amazonaws.com/ |

     
| Device | URL |
|--|--|
| mobile | http://estartandodevs-site.s3-website-sa-east-1.amazonaws.com/ |

Not what you expected? Are your scores flaky? [Run Lighthouse on Foo](https://www.foo.software/lighthouse)
If scores continue to be inconsistent consider [running all audits on Foo](https://github.com/foo-software/lighthouse-check-action#usage-foos-automated-lighthouse-check-api)

File: src/components/Courses/index.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        <img
+          src="eye.svg"
+          alt="Exibir módulos"
+          onClick={() => setExhibition(!exhibition)}
+        />
+      </S.ModulesTextContainer>
+      <S.ListModulesContainer>
+        {exhibition &&
+          literalObjectCall.courseModules.map((item, index) => (
+            <p key={index}>- {item}</p>
```
Comment (jonataspinto) [inline src/components/Courses/index.tsx:37]: Não entendi pq o lint não reclamou. Pode ajeitar @yagovaluchedevs ?

File: src/components/Courses/index.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import { Typography } from '..';
+import { COURSE_DATA } from '../../mocks/course';
+import * as S from './styles';
+
+type PropsCourse = {
+  course: 'frontEnd' | 'backEnd' | 'design';
+};
+
+const Courses = ({ course }: PropsCourse) => {
+  const [exhibition, setExhibition] = useState(false);
```
Comment (jonataspinto) [inline src/components/Courses/index.tsx:11]: Poderíamos usar query params mais pra frente, assim quando compartilhar, já vai com o box aberto.

File: public/eye.svg [hunk 9]
```diff
@@ -0,0 +1,29 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (jonataspinto) [inline public/eye.svg:1]: Esqueci de por esse no clodnary.

**Pull request merged**